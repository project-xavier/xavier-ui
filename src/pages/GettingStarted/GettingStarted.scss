.pf-c-empty-state > .dropzone {
  margin-top: var(--pf-c-empty-state--c-button--MarginTop);
}
