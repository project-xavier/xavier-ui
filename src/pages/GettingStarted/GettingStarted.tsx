import React from 'react';
import { Link } from 'react-router-dom';
import {
    EmptyStateBody,
    EmptyState,
    Title,
    Bullseye,
    EmptyStateVariant
} from '@patternfly/react-core';
import ProcessImprovementSvg from '../../PresentationalComponents/Icons/process-improvement_svg';
import './GettingStarted.scss';
import ReportsPage from '../../PresentationalComponents/ReportsPage/ReportsPage';
import { User } from '../../models';
import { Redirect } from 'react-router';

interface StateToProps {
    user: User | null;
}

interface DispatchToProps {
}

export interface Props extends StateToProps, DispatchToProps {
};

interface State {
};

class GettingStarted extends React.Component<Props, State> {

    constructor(props: Props) {
        super(props);
    }

    public render() {
        const { user } = this.props;

        if (user && !user.firstTimeCreatingReports) {
            return <Redirect to="/reports" />;
        }

        return (
            <ReportsPage>
                <Bullseye>
                    <EmptyState variant={ EmptyStateVariant.full }>
                        <div>
                            <ProcessImprovementSvg height="80px" className="pf-c-empty-state__icon" style={ { opacity: 0.6 } } />
                        </div>
                        <Title size="lg">Let Red Hat Migration Analytics suggest ways to optimize your environment</Title>
                        <EmptyStateBody>
                            See how you can optimize your virtual environment <br />
                            by uploading a Red Hat CloudForms generated Inventory file <br />
                            and then letting Red Hat Migration Analytics create reports <br />
                            showing how you can save money and optimize workloads <br />
                            whether by migrating virtual machines or <br />
                            migrating applications to Red Hat Enterprise Linux.
                        </EmptyStateBody>
                        <Link to={ '/reports/upload' } className="pf-c-button pf-m-primary">Get Started</Link>
                    </EmptyState>
                </Bullseye>
            </ReportsPage>
        );
    }
}

export default GettingStarted;
