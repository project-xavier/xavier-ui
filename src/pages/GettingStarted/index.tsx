import { GlobalState } from '../../models/state';
import GettingStarted from './GettingStarted';
import { connect } from 'react-redux';

const mapStateToProps = (state: GlobalState) => {
  const {
    userState: { user },
  } = state;
  return {
    user,
  };
};

const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(GettingStarted);
