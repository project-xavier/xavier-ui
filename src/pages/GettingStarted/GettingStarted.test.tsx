import React from 'react';
import { shallow } from 'enzyme';
import GettingStarted, { Props } from './GettingStarted';

const baseProps: Props = {
  user: {
    firstTimeCreatingReports: true,
  },
};

describe('GettingStarted', () => {
  it('expect to render', () => {
    const props = {
      ...baseProps,
    };

    const wrapper = shallow(<GettingStarted {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to redirect', () => {
    const props = {
      ...baseProps,
      user: {
        ...baseProps.user,
        firstTimeCreatingReports: false,
      },
    };

    const wrapper = shallow(<GettingStarted {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
