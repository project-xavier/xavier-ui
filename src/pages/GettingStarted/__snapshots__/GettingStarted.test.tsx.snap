// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GettingStarted expect to redirect 1`] = `
<Redirect
  push={false}
  to="/reports"
/>
`;

exports[`GettingStarted expect to render 1`] = `
<ReportsPage>
  <Bullseye>
    <EmptyState
      variant="full"
    >
      <div>
        <Icon
          className="pf-c-empty-state__icon"
          height="80px"
          style={
            Object {
              "opacity": 0.6,
            }
          }
        />
      </div>
      <Title
        size="lg"
      >
        Let Red Hat Migration Analytics suggest ways to optimize your environment
      </Title>
      <EmptyStateBody>
        See how you can optimize your virtual environment 
        <br />
        by uploading a Red Hat CloudForms generated Inventory file 
        <br />
        and then letting Red Hat Migration Analytics create reports 
        <br />
        showing how you can save money and optimize workloads 
        <br />
        whether by migrating virtual machines or 
        <br />
        migrating applications to Red Hat Enterprise Linux.
      </EmptyStateBody>
      <Link
        className="pf-c-button pf-m-primary"
        replace={false}
        to="/reports/upload"
      >
        Get Started
      </Link>
    </EmptyState>
  </Bullseye>
</ReportsPage>
`;
